### PR TITLE
Implement feature of 	Pullup Plugin: Add stop and restart methods

### PR DESCRIPTION
**Pull request by giiska**
Described in https://github.com/airyland/vux/issues/233

### CODE REVIEW

**Comment by airyland** [comment]
Thanks. A small problem `var self = this;` is not necessary. Could you remove the related lines and update the PR? :smiley:

**Comment by giiska** [comment]
@airyland done

File: build/cmd/plugins/pullup.js [hunk 1]
```diff
@@ -174,6 +174,20 @@ Util.extend(PullUp, Base, {
 		self.isLoading = false;
 		self._changeStatus("up");
 		return self;
+	},
+	stop: function() {
+		var xscroll = self.xscroll;
```
Comment (airyland) [inline build/cmd/plugins/pullup.js]: self => this

**Comment by giiska** [comment]
@airyland 😅  fixed

**Comment by airyland** [comment]
Still some left... `var xscroll = self.xscroll;` :sweat_smile:

**Comment by giiska** [comment]
Yes still some left in `complete` method. But the code of `complete` method is not committed by this pr. So I think its not necessary to rewrite it in this pr.

**Comment by airyland** [comment]
Yes, not necessary for other function , but as I can see, there are some left in `stop` function you wrote.

**Comment by giiska** [comment]
Oh I see, I forgot to build. Sorry for my careless.

**Comment by airyland** [comment]
Done publishing as v3.1.5

**Comment by lichunqiang** [comment]
squash into one commit will be better, It's hard to find the changes quickly...

File: src/plugins/pullup.js [hunk 2]
```diff
@@ -175,7 +175,7 @@ Util.extend(PullUp, Base, {
 		return self;
 	},
 	stop: function() {
-		var xscroll = self.xscroll;
+		var xscroll = this.xscroll;
```
Comment (lichunqiang) [inline src/plugins/pullup.js:178]: 为啥要这么做，下面没有调用这个变量啊

**Comment by airyland** [comment]
@lichunqiang You r right, should be more strict with PR commits.

File: src/plugins/pullup.js (hunk 2 above)
Comment (giiska) [inline src/plugins/pullup.js:178]: @lichunqiang   是啊。确实不该直接复制原作者的代码，他的代码如此：
https://github.com/bammoo/xscroll/blob/2677fb350f3c00d6496b9ea00000708eafbecde6/src/plugins/pullup.js#L172

Comment (lichunqiang) [inline src/plugins/pullup.js:178]: LOL

**Pull request merged**